### PR TITLE
Make `derivable_impls` machine applicable

### PR DESCRIPTION
**Pull request by kraktus**
changelog: [`derivable_impls`]: Now machine applicable

### CODE REVIEW

File: tests/ui/derivable_impls.stderr [hunk 1]
```diff
@@ -23,10 +27,14 @@ LL | |     }
 LL | | }
    | |_^
    |
-   = help: try annotating `TupleDefault` with `#[derive(Default)]`
+   = help: Remove it
+help: and instead derive it
+   |
+LL | #[derive(Default)]
+   |
```
Comment (kraktus) [inline tests/ui/derivable_impls.stderr:34]: Since the `Span` is 0-length long, the `struct XXX` part is not shown. I don't think this is a big issue now that it can be fixed automatically, but didn't find how to fix it in case it's a deal breaker.

**Review by xFrednet — COMMENTED** [review]
This looks good to me, I have a small NIT about the error message, but everything else is perfect :upside_down_face: 

Sorry, that the review took so long, my week has been very chaotic. Next review should be faster :upside_down_face:

File: clippy_lints/src/derivable_impls.rs [hunk 2]
```diff
@@ -100,15 +101,28 @@ impl<'tcx> LateLintPass<'tcx> for DerivableImpls {
...
                         "this `impl` can be derived",
-                        None,
-                        &format!("try annotating `{}` with `#[derive(Default)]`", path_string),
+                        |diag| {
+                            diag.span_suggestion_hidden(
+                                item.span,
+                                "remove it",
+                                String::new(),
+                                Applicability::MachineApplicable
+                            );
```
[the remark this answers is not in the document]
Comment (kraktus) [reply clippy_lints/src/derivable_impls.rs]: No need to apologise for the time taken everyone has the right to have a life! Addressed thanks

**Comment by xFrednet** [comment]
This version looks good to me, but you need to bless the test one more time, after fixing the typo :upside_down_face:

**Comment by kraktus** [comment]
Thank you, I don't know for sure the commit policy on the repo (squash everything in the end/no force push during review/...) so haven't cleaned the history (yet if needed)

**Comment by xFrednet** [comment]
We have a no merge commit policy. For the number of commits and squashing, we often go by feel. If the commits are structured well with good message or just one fix commit, we usually just merge it as it is. In this case, it would be cool, if you could squash them into one or two commits.

A small thing about squashing a rebasing is that GitHub doesn't work too well with it. So, after the first review, it's typically better to wait until the end for a rebase if you want to. You can of course also ask the reviewer. With small adjustments like this one, I often squash the change directly to the last commit and force push.

In short, we don't have a set in stone policy ^^. It would be cool if you could clean up the history, and then we can merge it :upside_down_face:

**Review by xFrednet — APPROVED** [review]

**Comment by xFrednet** [comment]
Perfect, this looks good to me! Thank you very much for this addition and the swift responses. :upside_down_face: 

@bors r+
